rust/lance/src/dataset/scanner.rs

```diff
@@ -689,11 +777,9 @@ impl Scanner {
         Ok(plan)
     }
 
-    // KNN search execution node with optional prefilter
-    async fn knn(
-        &self,
-        prefilter: Option<Arc<dyn PhysicalExpr>>,
-    ) -> Result<Arc<dyn ExecutionPlan>> {
+    // ANN/KNN search execution node with optional prefilter
```

Comment:
is filter plan still optional? Should we update the docstring

Reply:
Yes, the `FilterPlan` can be "no filter"